twisted/internet/asyncioreactor.py

```diff
@@ -0,0 +1,261 @@
+# -*- test-case-name: twisted.test.test_internet -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+asyncio-based reactor implementation.
+"""
+
+from __future__ import absolute_import, division
+
+import errno
+
+from zope.interface import implementer
+
+from twisted.logger import Logger
+from twisted.internet.base import DelayedCall
+from twisted.internet.posixbase import (PosixReactorBase, _NO_FILEDESC,
+                                        _ContinuousPolling)
+from twisted.python.log import callWithLogger
+from twisted.internet.interfaces import IReactorFDSet
+
+try:
+    from asyncio import new_event_loop
+except ImportError:
+    raise ImportError("Requires asyncio.")
+
+
+
+class _DCHandle(object):
+    """
+    Wrapper for L{asyncio.Handle} to be used by
```

Comment:
It looks like the point of this class is to provide a single reference the `DelayedCall`'s callbacks can close over that can resolve to arbitrary asyncio handles.  Maybe a better docstring would be something like:

```
    Wraps ephemeral L{asyncio.Handle} instances.  Callbacks can close
    over this and use it as a mutable reference to asyncio Handles.

    @ivar handle: The current L{asyncio.Handle}
```